source/common/router/BUILD

```diff
@@ -185,6 +185,29 @@ envoy_cc_library(
     ],
 )
 
+envoy_cc_library(
+    name = "filter_config_discovery_lib",
```

Comment:
Why is this part of router? I think logically HTTP filter management is part of HCM.

Reply:
Should I move it out of `include` tree then? Or move it to `envoy/http`? I'm not sure where to hoist it to.

Reply:
I'd also prefer to have a common definition for networking filters as well, but I wanted to cut a small slice first. If you have a suggestion on where to place this definition, I'll gladly take it.

Reply:
I think network/HTTP filter sharing is a really good point. I would be tempted to suggest `source/common/filter` (and `include/envoy/filter`. We could also potentially move some of the registry code there. But, that also leads to another consideration; do you want to generalize this to extensions beyond filters at any point? Filters are really just extensions that happen to be used in a few places like network/HTTP/listener.

Reply:
Yes, I would like to generalize this to extension config discovery service. Should we change the API to be ExtensionConfigDS?

Reply:
Yeah, I reckon we should do that. It's best to have clear terminology, and the Venn diagram has filters as a strict subset of extensions.

Reply:
SGTM. I'll update API definitions and also place the header in an appropriate place (with some templating).